kernel/shop/customerorderview.php

```diff
@@ -16,6 +16,13 @@
 $tpl = eZTemplate::factory();
 
 $Email = urldecode( $Email );
+
+// workaround because it seems not possible to get an urlencoded "+" character accross $Params
+if ( $http->hasGetVariable( "email" ) )
+{
+    $Email = $http->getVariable( "email" );
+}
+
```

Comment:
I'm not very happy with that patch, because the real problem lies in $Params (and a missing `urlencode()`). ;)